go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -197,7 +198,11 @@ type BaseSecondaryNetworkController struct {
 }
 
 func (oc *BaseSecondaryNetworkController) Reconcile(netInfo util.NetInfo) error {
-	return util.ReconcileNetInfo(oc.ReconcilableNetInfo, netInfo)
+	err := util.ReconcileNetInfo(oc.ReconcilableNetInfo, netInfo)
+	if err != nil {
+		return err
+	}
+	return oc.addAllPendingPods()
```

Comment:
Just note that this Reconcile might be called when something else on the NetInfo changed other than the NADs. So you might be calling `addAllPendingPods` when there is no need.
If you are fine with that, ok.

Other option is to compare the NADs in the new NetInfo against the old NetInfo before `ReconcileNetInfo, and just call `addAllPendingPods` if those specifically changed. That makes `ReconcileNetInfo` a point of no return, since further comparison would see no change: if `addAllPendingPods` fails, then the only thing you could do is log.

There is no mandate, but the second approach is the one we have followed elsewhere.

Reply:
Done.
Updated the PR to handle the cluster-manager ctrl too.